### ISSUE TITLE
SQS: rare IOException logged when listening for messages

### ISSUE THREAD

**Issue by zlangbert** [aws/aws-sdk-java-v2#522]
I have set up an SQS message receive loop using long polling. We noticed an IOException being thrown very rarely when listing for messages (see below). This can happen whether messages are being received or not. The client continues to work, and messages can still be received. This may just be due to general network hiccups, but I wanted to open an issue on this just in case it pointed to a larger problem. 

```
2018-06-05 19:34:46.106 WARN 99 --- [tyEventLoop-0-4] io.netty.channel.DefaultChannelPipeline : An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
java.io.IOException: Connection reset by peer 
	at sun.nio.ch.FileDispatcherImpl.read0(Native Method) ~[na:1.8.0_151] 
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39) ~[na:1.8.0_151] 
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[na:1.8.0_151] 
	at sun.nio.ch.IOUtil.read(IOUtil.java:192) ~[na:1.8.0_151] 
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380) ~[na:1.8.0_151] 
	at io.netty.buffer.PooledUnsafeDirectByteBuf.setBytes(PooledUnsafeDirectByteBuf.java:288) ~[netty-buffer-4.1.22.Final.jar!/:4.1.22.Final] 
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1108) ~[netty-buffer-4.1.22.Final.jar!/:4.1.22.Final] 
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:345) ~[netty-transport-4.1.22.Final.jar!/:4.1.22.Final] 
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:131) ~[netty-transport-4.1.22.Final.jar!/:4.1.22.Final] 
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:645) [netty-transport-4.1.22.Final.jar!/:4.1.22.Final] 
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:580) [netty-transport-4.1.22.Final.jar!/:4.1.22.Final] 
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:497) [netty-transport-4.1.22.Final.jar!/:4.1.22.Final] 
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:459) [netty-transport-4.1.22.Final.jar!/:4.1.22.Final] 
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:886) [netty-common-4.1.22.Final.jar!/:4.1.22.Final] 
	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_151]
```

## Steps to Reproduce (for bugs)

Long poll for messages in a loop. We see a handful of these a day with around a dozen threads listening for messages. One other thing to note is we have multiple threads using the same client in parallel listening for messages.

## Your Environment

* AWS Java SDK version used: `2.0.0-preview-10`
* JDK version used: 
```
openjdk version "1.8.0_151"
OpenJDK Runtime Environment (IcedTea 3.6.0) (Alpine 8.151.12-r0)
OpenJDK 64-Bit Server VM (build 25.151-b12, mixed mode)
```
* Operating System and version: `Alpine Linux 3.7`

**Comment by dagnir** [comment]
Thanks for reporting! As noted, this is probably due to general network hiccups. However, at minimum we should not be letting exceptions get past the end of the pipeline and provide a better error message.

**Comment by zlangbert** [comment]
That's what I was thinking as well, just some cleaner error handling so people aren't concerned. Thank you!

**Comment by mspiegle** [comment]
I added a comment to the DynamoDB version of this problem (#660).  It looks like the SDK keeps TCP connections open for efficient re-use, and the AWS server eventually closes them using a RST packet.  This seems to be part of the normal lifecycle of making a request to AWS and not related to network hiccups.  The strange thing is that I always see the connections stay open and eventually get closed with a reset, but I don't always get the exception.  Maybe this does point to a bug in certain situations?

**Comment by zoewangg** [comment]
Thank you for reporting. This has been fixed in #834.

**Comment by zoewangg** [comment]
The fix has been released as part of `2.1.0`. Closing the issue.
